go/engine/buffered_identify_ui.go

```diff
@@ -0,0 +1,243 @@
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, w := range b.proofChecks {
+		var err error
+		if w.social {
+			err = b.raw.FinishSocialProofCheck(w.p, w.l)
+		} else {
+			err = b.raw.FinishWebProofCheck(w.p, w.l)
+		}
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, c := range b.cryptocurrency {
+		err = b.raw.DisplayCryptocurrency(c)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (b *bufferedIdentifyUI) flushCleanup() {
+	b.start = nil
+	b.proofChecks = nil
+	b.cryptocurrency = nil
+	b.bufferedMode = false
+	b.launchNetworkChecks = nil
+	b.keys = nil
+	b.lastTrack = nil
+	b.userCard = nil
+}
+
+func (b *bufferedIdentifyUI) FinishWebProofCheck(p keybase1.RemoteProof, l keybase1.LinkCheckResult) error {
+	b.Lock()
+	defer b.Unlock()
+	b.proofChecks = append(b.proofChecks, proofCheck{false, p, l})
+	return b.flush(l.BreaksTracking)
+}
+
+func (b *bufferedIdentifyUI) FinishSocialProofCheck(p keybase1.RemoteProof, l keybase1.LinkCheckResult) error {
+	b.Lock()
+	defer b.Unlock()
+	b.proofChecks = append(b.proofChecks, proofCheck{true, p, l})
+	return b.flush(l.BreaksTracking)
+}
+
+func (b *bufferedIdentifyUI) Confirm(o *keybase1.IdentifyOutcome) (keybase1.ConfirmResult, error) {
+	b.Lock()
+	defer b.Unlock()
+	if b.bufferedMode {
+		b.G().Log.Debug("| bufferedIdentifyUI#Confirm: suppressing output")
+		b.suppressed = true
+		return b.confirmIfSuppressed, nil
+	}
+	b.G().Log.Debug("| bufferedIdentifyUI#Confirm: enabling output")
+	b.flush(true)
+	return b.raw.Confirm(o)
+}
+
+func (b *bufferedIdentifyUI) DisplayCryptocurrency(c keybase1.Cryptocurrency) error {
+	b.Lock()
+	defer b.Unlock()
+	b.cryptocurrency = append(b.cryptocurrency, c)
+	return b.flush(false)
+}
+
+func (b *bufferedIdentifyUI) DisplayKey(k keybase1.IdentifyKey) error {
+	b.Lock()
+	defer b.Unlock()
+	b.keys = append(b.keys, k)
+	return b.flush(false)
+}
+
+func (b *bufferedIdentifyUI) ReportLastTrack(s *keybase1.TrackSummary) error {
+	b.Lock()
+	defer b.Unlock()
+	b.lastTrack = &s
+	return b.flush(false)
+}
+
+func (b *bufferedIdentifyUI) LaunchNetworkChecks(i *keybase1.Identity, u *keybase1.User) error {
+	b.Lock()
+	defer b.Unlock()
+	b.launchNetworkChecks = &launchNetworkChecks{i, u}
+	return b.flush(i.BreaksTracking)
+}
+
+func (b *bufferedIdentifyUI) DisplayTrackStatement(s string) error {
+	return b.raw.DisplayTrackStatement(s)
+}
+
+func (b *bufferedIdentifyUI) DisplayUserCard(c keybase1.UserCard) error {
+	b.Lock()
+	defer b.Unlock()
+	b.userCard = &c
+	return b.flush(false)
+}
+
+func (b *bufferedIdentifyUI) ReportTrackToken(t keybase1.TrackToken) error {
+	b.Lock()
+	defer b.Unlock()
+	if b.suppressed {
+		return nil
+	}
+	return b.raw.ReportTrackToken(t)
+}
+
+func (b *bufferedIdentifyUI) Finish() error {
+	b.Lock()
+	defer b.Unlock()
+	if b.suppressed {
+		b.G().Log.Debug("| bufferedIdentifyUI#Finish: suppressed")
+		return nil
+	}
+	b.G().Log.Debug("| bufferedIdentifyUI#Finish: went through to UI")
```

Comment:
it can't hurt, but i don't think it will do anything, since the `flush` in `Confirm` covered this case.